tests/unit/test_backup_observer.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Synapse backup observer unit tests."""
+
+
+from secrets import token_hex
+from unittest.mock import MagicMock
+
+import boto3
+import ops
+import pytest
+from ops.testing import Harness
+
+import backup_observer
+
+
+def test_on_s3_credentials_changed_correct(harness: Harness, monkeypatch: pytest.MonkeyPatch):
```

Comment:
Do you think this and the next 3 tests could be just one using pytest.mark.parametrize?

Reply:
Renamed integration from s3-backup to backup as suggested by @cbartz

Reply:
Joined. Not sure which option looks better (now it looks a bit like Go)